core/infra/config/peer_service_with_json.cpp

```diff
@@ -117,39 +123,65 @@ std::vector<std::string> PeerServiceConfig::getIpList() {
   return ret_ips;
 }
 
+
+// check are broken? peer
+void PeerServiceConfig::checkBrokenPeer( const std::string& ip ) {
+    if (!isExistIP(ip)) return;
+    auto check_peer_it = findPeerIP( ip );
+    if ( !connection::iroha::PeerService::Sumeragi::ping( ip )) {
+        if( check_peer_it->getTrustScore() < 0.0 ) {
+            toIssue_removePeer( check_peer_it->getPublicKey() );
+        } else {
+            toIssue_distructPeer( check_peer_it->getPublicKey() );
```

Comment:
toIssue_distructPeer means to Issue "Think to distrust" or "suspicous".
toIssue_disturctPeer does to issue transaction that decrease trustScore from target Peer.